builtin/credential/approle/path_role.go

```diff
@@ -548,8 +937,49 @@ Overrides secret_id_ttl role option when supplied. May not be longer than role's
 					Description: "Accessor of the SecretID",
 				},
 			},
-			Callbacks: map[logical.Operation]framework.OperationFunc{
-				logical.UpdateOperation: b.pathRoleSecretIDAccessorLookupUpdate,
+			Operations: map[logical.Operation]framework.OperationHandler{
+				logical.UpdateOperation: &framework.PathOperation{
+					Callback: b.pathRoleSecretIDAccessorLookupUpdate,
+					Responses: map[int][]framework.Response{
+						http.StatusOK: {{
+							Description: "OK",
+							Fields: map[string]*framework.FieldSchema{
+								"secret_id_accessor": {
+									Type:        framework.TypeString,
+									Description: "Accessor of the secret ID",
+								},
+								"secret_id_ttl": {
+									Type:        framework.TypeDurationSecond,
+									Description: "Duration in seconds after which the issued secret ID expires.",
+								},
+								"secret_id_num_uses": {
+									Type:        framework.TypeInt,
+									Description: "Number of times a secret ID can access the role, after which the secret ID will expire.",
+								},
+								"creation_time": {
+									Type: framework.TypeTime,
+								},
+								"expiration_time": {
+									Type: framework.TypeTime,
+								},
+								"last_updated_time": {
+									Type: framework.TypeTime,
+								},
+								"metadata": {
+									Type: framework.TypeMap,
+								},
+								"cidr_list": {
+									Type:        framework.TypeCommaStringSlice,
+									Description: "List of CIDR blocks enforcing secret IDs to be used from specific set of IP addresses. If 'bound_cidr_list' is set on the role, then the list of CIDR blocks listed here should be a subset of the CIDR blocks listed on the role.",
+								},
+								"token_bound_cidrs": {
+									Type:        framework.TypeCommaStringSlice,
+									Description: "List of CIDR blocks. If set, specifies the blocks of IP addresses which can use the returned token. Should be a subset of the token CIDR blocks listed on the role, if any.",
```

Comment:
this is a lot by hand.... can we use reflection at all here to get type info?

Reply:
The rest of the code in the file is all handcrafted so it would look a bit weird. However, I'd be very eager to find some way to automate the process of filling in the response information going forward. Any ideas would be welcome.